tests/test_resources/test_clusters/test_cluster.py

```diff
@@ -850,8 +850,12 @@ def test_cluster_run_within_cluster(self, cluster):
         res = remote_run("echo hello")
         exp = cluster.run_bash("echo hello")
 
-        assert res[0] == 0
-        assert res[1].strip() == exp[1].strip()
+        if cluster.config().get("resource_subtype") == "Cluster":
```

Comment:
I think `run_bash` _should_ return `(retcode, stdout, stderr)`, for both cluster and ondemand cluster, are you not seeing this?